boards: st: nucleo_h563zi: unsupported AES peripheral in device tree

### Describe the bug

Building the crypto sample for the nucleo_h563zi board results in a compilation error.

The [nucleo_h563zi board page](https://docs.zephyrproject.org/latest/boards/st/nucleo_h563zi/doc/index.html) lists `st,stm32-aes` compatible. But only the STM32H573 variant supports the AES peripheral (https://www.st.com/en/microcontrollers-microprocessors/stm32h563-573.html)

The `st,stm32-aes` should only be added to `stm32h573.dtsi`

### Regression

- [ ] This is a regression.

### Steps to reproduce

1. `cd samples/drivers/crypto`
2. create boards/nucleo_h563zi.overlay with `&aes { status = "okay"; };`
3. `west build --board nucleo_h563zi`

### Relevant log output

```shell
[135/163] Building C object zephyr/drivers/crypto/CMakeFiles/drivers__crypto.dir/crypto_stm32.c.obj
FAILED: [code=1] zephyr/drivers/crypto/CMakeFiles/drivers__crypto.dir/crypto_stm32.c.obj 
ccache C:\Workspace\rtos-zephyr-ws\zephyr-sdk-1.0.1\gnu\arm-zephyr-eabi\bin\arm-zephyr-eabi-gcc.exe -DCORE_CM33 -DHSE_VALUE=8000000 -DKERNEL -DK_HEAP_MEM_POOL_SIZE=0 -DPICOLIBC_LONG_LONG_PRINTF_SCANF -DSTM32H563xx -DUSE_FULL_LL_DRIVER -DUSE_HAL_DRIVER -D_POSIX_THREAD_SAFE_FUNCTIONS=200809L -D__LINUX_ERRNO_EXTENSIONS__ -D__PROGRAM_START -D__ZEPHYR_SUPERVISOR__ -D__ZEPHYR__=1 -IC:/Workspace/rtos-zephyr-ws/zephyr/samples/drivers/crypto/build/zephyr/include/generated/zephyr -IC:/Workspace/rtos-zephyr-ws/zephyr/include -IC:/Workspace/rtos-zephyr-ws/zephyr/samples/drivers/crypto/build/zephyr/include/generated -IC:/Workspace/rtos-zephyr-ws/zephyr/soc/st/stm32 -IC:/Workspace/rtos-zephyr-ws/zephyr/lib/libc/picolibc/include -IC:/Workspace/rtos-zephyr-ws/zephyr/soc/st/stm32/common/. -IC:/Workspace/rtos-zephyr-ws/zephyr/drivers -IC:/Workspace/rtos-zephyr-ws/zephyr/soc/st/stm32/stm32h5x/. -IC:/Workspace/rtos-zephyr-ws/zephyr/subsys/portability/posix/c_lib_ext/getopt -IC:/Workspace/rtos-zephyr-ws/example-application/include -IC:/Workspace/rtos-zephyr-ws/modules/hal/cmsis_6/CMSIS/Core/Include -IC:/Workspace/rtos-zephyr-ws/zephyr/modules/cmsis_6/. -IC:/Workspace/rtos-zephyr-ws/modules/hal/stm32/stm32cube/stm32h5xx/soc -IC:/Workspace/rtos-zephyr-ws/modules/hal/stm32/stm32cube/stm32h5xx/drivers/include -IC:/Workspace/rtos-zephyr-ws/modules/hal/stm32/stm32cube/common_ll/include -isystem C:/Workspace/rtos-zephyr-ws/zephyr/lib/libc/common/include -fdiagnostics-color=always -Wshadow -fno-strict-aliasing -Os -imacros C:/Workspace/rtos-zephyr-ws/zephyr/samples/drivers/crypto/build/zephyr/include/generated/zephyr/autoconf.h -fno-printf-return-value -fno-common -g -gdwarf-4 -fdiagnostics-color=always -mcpu=cortex-m33 -mthumb -mabi=aapcs -mfp16-format=ieee -mtp=soft --sysroot=C:/Workspace/rtos-zephyr-ws/zephyr-sdk-1.0.1/gnu/arm-zephyr-eabi/arm-zephyr-eabi -imacros C:/Workspace/rtos-zephyr-ws/zephyr/include/zephyr/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wdouble-promotion -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -fno-asynchronous-unwind-tables -ftls-model=local-exec -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=C:/Workspace/rtos-zephyr-ws/zephyr/samples/drivers/crypto=CMAKE_SOURCE_DIR -fmacro-prefix-map=C:/Workspace/rtos-zephyr-ws/zephyr=ZEPHYR_BASE -fmacro-prefix-map=C:/Workspace/rtos-zephyr-ws=WEST_TOPDIR -ffunction-sections -fdata-sections -specs=picolibc.specs -std=c17 -MD -MT zephyr/drivers/crypto/CMakeFiles/drivers__crypto.dir/crypto_stm32.c.obj -MF zephyr\drivers\crypto\CMakeFiles\drivers__crypto.dir\crypto_stm32.c.obj.d -o zephyr/drivers/crypto/CMakeFiles/drivers__crypto.dir/crypto_stm32.c.obj -c C:/Workspace/rtos-zephyr-ws/zephyr/drivers/crypto/crypto_stm32.c
In file included from C:/Workspace/rtos-zephyr-ws/zephyr/drivers/crypto/crypto_stm32.c:18:
C:/Workspace/rtos-zephyr-ws/zephyr/drivers/crypto/crypto_stm32_priv.h:26:9: error: unknown type name 'CRYP_HandleTypeDef'; did you mean 'CRC_HandleTypeDef'?
   26 |         CRYP_HandleTypeDef hcryp;
      |         ^~~~~~~~~~~~~~~~~~
      |         CRC_HandleTypeDef
C:/Workspace/rtos-zephyr-ws/zephyr/drivers/crypto/crypto_stm32_priv.h:14:24: error: unknown type name 'CRYP_ConfigTypeDef'; did you mean 'RNG_ConfigTypeDef'?
   14 | #define crypt_config_t CRYP_ConfigTypeDef
      |                        ^~~~~~~~~~~~~~~~~~
```

### Impact

Annoyance – Minor irritation; no significant impact on usability or functionality.

### Environment

- Windows 11
- Zephyr version: 4.4.99, build: v4.4.0-607-g599b67f7da25, commit 599b67f7da2
- Zephyr SDK version: 1.0.1

### Additional Context


boards/nucleo_h563zi.overlay
```devicetree
&aes {
	status = "okay";
};
```

https://github.com/zephyrproject-rtos/zephyr/blob/599b67f7da25d4f4af66886a169f007e3ab866d4/dts/arm/st/h5/stm32h562.dtsi#L526-L533

--- a/dts/arm/st/h5/stm32h533.dtsi
+++ b/dts/arm/st/h5/stm32h533.dtsi
@@ -5,6 +5,7 @@
  */
 
 #include <st/h5/stm32h523.dtsi>
+#include <st/h5/stm32h5_crypt.dtsi>
 
 / {
 	soc {

--- a/dts/arm/st/h5/stm32h562.dtsi
+++ b/dts/arm/st/h5/stm32h562.dtsi
@@ -524,15 +524,6 @@
 			};
 		};
 
-		aes: aes@420c0000 {
-			compatible = "st,stm32-aes";
-			reg = <0x420c0000 0x400>;
-			clocks = <&rcc STM32_CLOCK(AHB2, 16)>;
-			resets = <&rctl STM32_RESET(AHB2, 16)>;
-			interrupts = <116 0>;
-			status = "disabled";
-		};
-
 		rng: rng@420c0800 {
 			nist-config = <0xf00e00>;
 			health-test-config = <0x6a91>;

--- a/dts/arm/st/h5/stm32h573.dtsi
+++ b/dts/arm/st/h5/stm32h573.dtsi
@@ -5,6 +5,7 @@
  */
 
 #include <st/h5/stm32h563.dtsi>
+#include <st/h5/stm32h5_crypt.dtsi>
 
 / {
 	soc {

--- a/dts/arm/st/h5/stm32h5_crypt.dtsi
+++ b/dts/arm/st/h5/stm32h5_crypt.dtsi
@@ -1,0 +1,18 @@
+/*
+ * Copyright (c) 2026 STMicroelctronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/ {
+	soc {
+		aes: aes@420c0000 {
+			compatible = "st,stm32-aes";
+			reg = <0x420c0000 0x400>;
+			clocks = <&rcc STM32_CLOCK(AHB2, 16)>;
+			resets = <&rctl STM32_RESET(AHB2, 16)>;
+			interrupts = <116 0>;
+			status = "disabled";
+		};
+	};
+};

--- a/dts/arm/st/h5/stm32h5f4.dtsi
+++ b/dts/arm/st/h5/stm32h5f4.dtsi
@@ -5,6 +5,7 @@
  */
 
 #include <st/h5/stm32h5e4.dtsi>
+#include <st/h5/stm32h5_crypt.dtsi>
 
 / {
 	soc {

--- a/dts/arm/st/h5/stm32h5f5.dtsi
+++ b/dts/arm/st/h5/stm32h5f5.dtsi
@@ -5,6 +5,7 @@
  */
 
 #include <st/h5/stm32h5e5.dtsi>
+#include <st/h5/stm32h5_crypt.dtsi>
 
 / {
 	soc {

